taxcalc/tests/test_functions.py

```diff
@@ -0,0 +1,40 @@
+"""
+Tests for Tax-Calculator functions.py file.
+"""
+# CODING-STYLE CHECKS:
+# pep8 --ignore=E402 test_functions.py
+# pylint --disable=locally-disabled test_functions.py
+# (when importing numpy, add "--extension-pkg-whitelist=numpy" pylint option)
+
+import os
+import sys
+import re
+CUR_PATH = os.path.abspath(os.path.dirname(__file__))
+
+
+def test_function_args_usage():
+    """
+    Checks each function argument in functions.py for use in its function body.
+    """
+    funcfilename = '{}/../functions.py'.format(CUR_PATH)
+    with open(funcfilename, 'r') as funcfile:
```

Comment:
@talumbau said:
> To go down this path, I would suggest making use of the inspect module, documented here - in particular, inspect.getsource().

Thanks for the suggestion, but after looking at the inspect module, it wasn't obvious how its use would simplify matters or make the new test more robust.  In particular, getsource() requires the name of the function whose source code you want.  So, it would seem, the functions.py text would have to be parsed for a list function names.  In addition, because I don't want the full code of each function (but just the code before the return statement), I would still have to manipulate the results from inspect.getsource() in order to remove the return statement.

It may turn out that your concerns are well founded, and I'll have to do further work on this new test.  But why don't we take a wait-and-see approach?